sunscreen_backend/src/transforms/insert_relinearizations.rs

```diff
@@ -64,13 +64,50 @@ pub fn apply_insert_relinearizations(ir: &mut FheProgram) {
         Ok::<_, Infallible>(transforms)
     })
     .unwrap();
+
+    #[cfg(feature = "debugger")]
+    {
+        let mut group_updates: Vec<(u64, u64)> = vec![];
+        for i in ir.graph.node_indices() {
+            group_updates.push((
+                ir.graph.node_weight(i).unwrap().group_id,
+                i.index().try_into().unwrap(),
+            ));
+        }
+        for (g, i) in group_updates {
+            group_insert_recursive(&g, i, ir);
```

Comment:
I had made some comments on your other review to lazily compute the display graph when the user makes a REST call so you don't have a bunch of extra data structures floating around that you have to keep up to date.

I still strongly think we should do that and remove this code. The only thing this compiler transform should do WRT to groups is ensure the relin node has the same group id as the node that comes before it.

Reply:
Actually, you should be able to just remove this code?